db/savedata.go

```diff
@@ -142,3 +142,64 @@ func DeleteSessionSaveData(uuid []byte, slot int) error {
 
 	return nil
 }
+
+func RetrieveAccountEggs(uuid []byte) ([]defs.EggData, error) {
+	var accountEggs []defs.EggData
+
+	rows, err := handle.Query("SELECT uuid, gachaType, hatchWaves, timestamp FROM eggs WHERE owner = ?", uuid)
+	if err != nil {
+		return accountEggs, err
+	}
+
+	// For each row, we parse the raw data into an EggData and add it to the result
+	for rows.Next() {
+		var egg defs.EggData
+		err = rows.Scan(&egg.Id, &egg.GachaType, &egg.HatchWaves, &egg.Timestamp)
+		if err != nil {
+			return accountEggs, err
+		}
+
+		accountEggs = append(accountEggs, egg)
+	}
+
+	return accountEggs, nil
+}
+
+func UpdateAccountEggs(uuid []byte, eggs []defs.EggData) error {
+	tx, err := handle.Begin()
+	if err != nil {
+		return err
+	}
+
+	for _, egg := range eggs {
+		// TODO: find a fix to enforce encoding from body to EggData only if
+		// it respects the EggData struct so we can get rid of the test 
+		if egg.Id == 0 {
+			continue
+		}
+
+		_, err = tx.Exec(`INSERT INTO eggs (uuid, owner, gachaType, hatchWaves, timestamp) 
```

Comment:
have you thought about storing the whole `[]defs.EggData` as a binary blob (similar like other stuff in a db)? Probably we don't want to query over the `egg` params, so it may be the easiest and most flexible solution

Reply:
I think having the egg data defined in the db is better since it's gives us query access to fields we want to edit (only `hatchWaves` is relevant on updates)
The main reason we have to redo some endpoints/tables is because saved data is stored as a blob and we have to update the whole thing even if we are only changing a few fields. 
On a scale of eggs I don't think we would see a negative impact as it's so small, but it's the same principle, instead of updating only the `hatchWaves` field, we would be updating everything which is more data transferred.

I do agree that it would be easier and avoid the json parsing process, but I was told that the main issue was to reduce data transferred from server to db